File: Core/Entities/OrderAggregate/Order.cs
using Microsoft.EntityFrameworkCore;

namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public string CustomerEmail { get; set; }

        public Address DeliveryAddress { get; set; }

        public float SubTotal { get; set; }

        public float DeliveryCharges { get; set; }

        public IReadOnlyList<OrderItem> OrderItems { get; set; }

        public OrderStatus Status { get; set; }

        public PaymentStatus PaymentStatus { get; set; }

        public float GetTotal()
        {
            return SubTotal + DeliveryCharges;
        }
    }
}

File: client/src/app/shared/models/order.ts
import { IShoppingCartItem } from "./shoppingCart";

export interface IOrder{
  cartId: string;
  address: IAddress
  callbackUrl: string
}

export interface IAddress{
  firstName: string;
  lastName: string;
  street: string;
  city: string;
  state: string;
}

export interface ISummaryOrder {
  id: string;
  customerEmail: string;
  dateCreated: string;
  deliveryCharges: number;
  paymentStatus: string;
  status: string;
  subTotal: number;
  total: number;
  deliveryAddress: IAddress;
  orderItems: IOrderSummaryItem[];
}

export interface IOrderSummaryItem{
  productName: string;
  pictureUrl: string;
  quantity: number;
  price:number;
}
